File: Тренировка 1.0/03. Множества/I. Полиглоты/Полиглоты.cs
using System;
using System.Collections.Generic;

int N = int.Parse(Console.ReadLine());

HashSet<string> allLanguages = new HashSet<string>();
Dictionary<string, int> languagesMap = new Dictionary<string, int>();
for(int i = 0; i < N; i++)
{
    int M = int.Parse(Console.ReadLine());
    for(int j = 0; j < M; j++)
    {
        string language = Console.ReadLine();
        if(!languagesMap.ContainsKey(language)) languagesMap.Add(language, 0);
        languagesMap[language]++;
        allLanguages.Add(language);
    }
}

int onlyCount = 0;
foreach (var pair in languagesMap)
    if (pair.Value == N) onlyCount++;

Console.WriteLine(onlyCount);
foreach (var pair in languagesMap)
    if(pair.Value == N) Console.WriteLine(pair.Key);

Console.WriteLine(allLanguages.Count);
foreach (string language in allLanguages) Console.WriteLine(language);

File: Тренировка 1.0/03. Множества/I. Полиглоты/Полиглоты.cpp
#include <iostream>
#include <unordered_set>
#include <unordered_map>

int main()
{
    int N, M;
    std::cin >> N;
    std::string language;
    std::unordered_set<std::string> allLanguages;
    std::unordered_map<std::string, int> languagesMap;
    for (int i = 0; i < N; i++)
    {
        std::cin >> M;
        for (int j = 0; j < M; j++)
        {
            std::cin >> language;
            if (!languagesMap.count(language)) languagesMap.emplace(std::make_pair(language, 0));
            languagesMap[language]++;
            allLanguages.emplace(language);
        }
    }

    int onlyCount = 0;
    for(auto pair : languagesMap)
        if (pair.second == N) onlyCount++;

    std::cout << onlyCount << std::endl;
    for (auto pair : languagesMap)
        if (pair.second == N) std::cout << pair.first << std::endl;

    std::cout << allLanguages.size() << std::endl;
    for (auto language : allLanguages) std::cout << language << std::endl;
}